### PR TITLE
[MRG+1] Exclude to edf reader.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3876
Closes https://github.com/mne-tools/mne-python/issues/3890

I think I finally got it working. Let's wait until https://github.com/mne-tools/mne-python/pull/3895 gets merged and I'll rebase.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
LGTM, but the complexity of this reader makes me think we do need to add a test file for #3895, and maybe even a different one for this resampling issue, before merging these :(

**Comment by jaeilepp** [comment]
Agreed. I wonder if there's a tool available for creating edf files manually.

**Comment by agramfort** [comment]
@jaeilepp can you rebase + fix travis

@slasnista can you confirm it's working for you?

File: mne/io/edf/edf.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ class RawEDF(BaseRaw):
     annotmap : str | None
         Path to annotation map file containing mapping from label to trigger.
         Must be specified if annot is not None.
+    exclude : array of str
+        Channel names to exclude.
```
Comment (agramfort) [inline mne/io/edf/edf.py]: Add a note that this can help loading channels with higher sampling rates to avoid unnecessary resampling.

**Comment by agramfort** [comment]
also update what's new and to me we're good to go.

**Comment by jaeilepp** [comment]
Ready for review.

File: mne/io/edf/edf.py [hunk 2]
```diff
@@ -51,6 +51,9 @@ class RawEDF(BaseRaw):
     annotmap : str | None
         Path to annotation map file containing mapping from label to trigger.
         Must be specified if annot is not None.
+    exclude : array of str
```
Comment (Eric89GXL) [inline mne/io/edf/edf.py]: we usually call this `list of str`

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**